### ISSUE TITLE
warn if multibyte wordlist is attempted against NTLM hashes

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#2566]
It might not hurt for hashcat to emit a warning if NTLM is attempted with a multibyte wordlist.

limits.txt says:

> ##
> ## Hashing algorithms that internally use UTF-16 characters could in special cases lead to false negatives
> ##
> 
> The UTF-16 conversion implementation used within the kernel code is very elementary and for performance
> reasons does not respect all complicated encoding rules required to correctly convert, for instance, ASCII
> or UTF-8 to UTF-16LE (or UTF-16BE).
> 
> The implementation most likely fails with multi-byte characters, because we basically add a zero byte every
> second byte within the kernel conversion code.

With i18n, the rise of emoji, etc, this is becoming somewhat of a FAQ item in various forums and channels. Many casual users are not likely to read limits.txt, or even know that it exists (heck, I forget sometimes!)

This is assuming that multibyte could be detected during the initial dict/wordlist ingestion analysis.

**Comment by jsteube** [comment]
With the added support of true UTF8 to UTF16 conversion we now have support for this kind of wordlist. However, this is applied for all slow-hash kernel types and for fast-hash kernel types in pure kernel mode. It is not applied in fast-hash kernel type if in optimized kernel mode.

Actually I think now this message makes even more sense. It's very unlikely users that rarely use hashcat know about these fine differences.

The question is how we can detect a multibyte wordlist. Yes we can detect if a word in a wordlist is in multibyte encoding, but that's a very theoretical scenario. A clean wordlist that really only is encoded in UTF8 is rare. Most wordlists include a wild mix of ISO-8859-X, UTF8, UTF16 and whatever codepage. 

Scanning a file to detect the text encoding is actually a huge problem. The 'file' command on Linux deal with this and has tons of branches in order to do so, and still it fails because it can never be correct. Instead it really goes through every line and detects the encoding of every word (which can be multiple) and collect them and create a summary in the end. 

This could create more problems than it solves. Maybe it's easier to just print a warning message without scanning the wordlist. But then it would show up every time and from experience this kind of error message is it that the user "unsee" it after they saw it 10 times in a row and nothing happens.

**Comment by roycewilliams** [comment]
Yep, that totally makes sense. Hot take: if there are a few _easy_ ways to detect multibyte, they could be added and throw the warning, and assume that a few false positives are possible, and let the warning in limits.txt cover the rest.

**Comment by Chick3nman** [comment]
I agree that the warning message may get "unseen" through repeated usage but I think it's a good idea to have it regardless. We can include it as ADVICE level to allow heavy users to disable it if they do not wish to see it. Do we want to set a module flag for this and a generic warning? "One or more optimizers for this hash mode may break support for non-ascii characters" or something similar.

**Comment by roycewilliams** [comment]
That sounds like a good general solution!
